Add OOTB script

kobi86 opened the pull request:
@eifrach enjoy 

But please test it on linux VM first to make all is working

kobi86 committed:
Add OOTB script

kobi86 commented:
Hey @cwilkers, 
Can you please have a look at this and test it to make sure its working on linux VM :)

cwilkers commented:
Hi @kobi86 thanks for your contribution! I'll take a look.

cwilkers reviewed (changes requested):
First off, we could use some better documentation about the layout of this repository. The blogs directory is meant to hold any contributing code or YAML in support of specific published blogs, so this script, being part of the scripted windows blog, should belong under that same directory.

The code/topology directory is for generating DOT graphs from deployments and daemonsets in a namespace, and has nothing to do with Windows installs.

Please move the script to blogs/scripted-windows and adjust paths appropriately.

cwilkers commented on code/topology/OOTB_install.sh:
```diff
@@ -0,0 +1,40 @@
+PASS=$(echo "$OCPASS" | base64)
+
+sed -i.bak "s/<base64 URL>/$API/" $OCFILE
+sed -i.bak "s/<base64 user>/$USER/" $OCFILE
```
Multiple sed commands with the backup option will not preserve the original file, you will only get the file state after the second to last sed. If the goal is to make this idempotent, combine the sed substitutions into one block, and create a new file (no in-place).

Then you can change the test for existence of the .bak file to a test for existence of the populated credentials file, and instead of exiting, use that to skip all the questions and sed.

cwilkers commented on code/topology/OOTB_install.sh:
```diff
@@ -0,0 +1,40 @@
+sed -i.bak "s/<base64 URL>/$API/" $OCFILE
+sed -i.bak "s/<base64 user>/$USER/" $OCFILE
+sed -i.bak "s/<base64 password>/$PASS/" $OCFILE
+
```
You go through all the trouble of making the credential grab interactive, but you never create the secret in the cluster from the file...

kobi86 replied:
Talked, Fixed the .bak file

kobi86 committed:
Adjusted the script to fit blog

- Fixed bak overwirte
- Moved script file the right location
- Updated README file

kobi86 commented:
@cwilkers updated

cwilkers commented on blogs/scripted-windows/OOTB_install.sh:
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+OCFILE="./oc-creds.yaml"
+YAMLFILE="./scripted-windows/"
```
This is causing troubles because we are now running from the `scripted-windows` directory.

```suggestion
YAMLFILE="."
```

kobi86 replied:
Done

cwilkers commented on blogs/scripted-windows/OOTB_install.sh:
```diff
@@ -0,0 +1,38 @@
+sed -i.bak "s/<base64 URL>/$API/" $OCFILE
+sed -i "s/<base64 user>/$USER/" $OCFILE
+sed -i "s/<base64 password>/$PASS/" $OCFILE
+
```
We still are not applying the secret we just created...
```suggestion

kubectl apply -f $OCFILE
```

kobi86 replied:
Right, so in that case the command should be: 
kubectl apply -f $YAMLFILE/
This will publish the all folder and all the yaml's and will help clean the code more:)

cwilkers commented on code/topology/README.md:
```diff
@@ -10,3 +10,4 @@
 ## Usage
 
     ./hco_topo.sh | dot -Tpng -o topo.png
+
```
Not sure if this will work, may have to checkout this file from another branch to undo the trailing newline.
```suggestion
```

kobi86 replied:
Done

kobi86 commented:
@cwilkers Updated And fixed what you pointed

cwilkers commented:
Thanks again @kobi86 !
